### PR TITLE
[autoscaler][aws] Validate current state of subnet-specification

### PR DESCRIPTION
**Pull request by DmitriGekhtman**
## Why are these changes needed?

To correctly configure a security group, the logic of `bootstrap_aws` currently requires subnets to be specified `head_nodes` and `worker_nodes` fields.

This PR adds a test that validates that adding head_nodes and worker_nodes fields with subnet data to a multi-node-type config leads to a correct configuration of a security group.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(
 Test passes.

### CODE REVIEW

**Comment by DmitriGekhtman** [comment]
(for some reason, wasn't able to charles or allen as assignees)

File: python/ray/tests/aws/test_autoscaler_aws.py [hunk 1]
```diff
@@ -164,6 +166,122 @@ def test_fills_out_amis(iam_client_stub, ec2_client_stub):
...
+
+    # Confirming boostrap config does not currently touch available node types.
+    assert bootstrapped_config["available_node_types"] == bootstrapped_config[
+        "available_node_types"]
+
+    # Confirming head and worker subnet_ids are untouched
+    assert bootstrapped_config["head_node"]["SubnetIds"] ==\
+        config["head_node"]["SubnetIds"] ==\
+        config["worker_nodes"]["SubnetIds"] ==\
+        bootstrapped_config["worker_nodes"]["SubnetIds"]
```
Comment (xcharleslin) [inline python/ray/tests/aws/test_autoscaler_aws.py:266]:
```suggestion
        bootstrapped_config["worker_nodes"]["SubnetIds"] ==\
        [DEFAULT_SUBNET["SubnetId"]]
```

^ ?

File: python/ray/tests/aws/test_autoscaler_aws.py [hunk 2]
```diff
@@ -164,6 +166,122 @@ def test_fills_out_amis(iam_client_stub, ec2_client_stub):
...
+
+    # Confirming correct security group got filled for head and workers
+    sg_id = DEFAULT_SG["GroupId"]
+    assert bootstrapped_config["head_node"]["SecurityGroupIds"] == [sg_id]
+    assert bootstrapped_config["worker_nodes"]["SecurityGroupIds"] == [sg_id]
```
Comment (xcharleslin) [inline python/ray/tests/aws/test_autoscaler_aws.py:271]: Huh, interesting - are we still relying on deprecated `head_node` and `worker_node_type` to propagate SecurityGroupIds?

**Comment by xcharleslin** [comment]
Btw can you link me to the premerge run of this test? I'm not familiar with the OSS CI setup :p

File: python/ray/tests/aws/test_autoscaler_aws.py (hunk 2 above)
Comment (DmitriGekhtman) [reply python/ray/tests/aws/test_autoscaler_aws.py:271]: Yep -- this also happens in `aws/config.py`.

**Comment by DmitriGekhtman** [comment]
> Btw can you link me to the premerge run of this test? I'm not familiar with the OSS CI setup :p
Yyeah -- not the most intuitive thing to find. 
https://travis-ci.com/github/ray-project/ray/jobs/493000203#L3193

File: python/ray/tests/aws/test_autoscaler_aws.py (hunk 1 above)
Comment (DmitriGekhtman) [reply python/ray/tests/aws/test_autoscaler_aws.py:266]: sure! 
It's implicit in the way `config["head_node"]` was defined, but better to be explicit.

File: python/ray/tests/aws/test_autoscaler_aws.py (hunk 2 above)
Comment (xcharleslin) [reply python/ray/tests/aws/test_autoscaler_aws.py:272]: I see. Tangential to this PR, what other behaviour do we need to update in aws/config.py for available_node_types? Is it just subnets, or?

**Review by xcharleslin — APPROVED** [review]
Thank you for writing this test in OSS :)

**Comment by DmitriGekhtman** [comment]
@simon-mo could you merge this?

**Comment by DmitriGekhtman** [comment]
Let's just wait for tests to finish..

File: python/ray/tests/aws/test_autoscaler_aws.py [hunk 3]
```diff
@@ -164,6 +166,123 @@ def test_fills_out_amis(iam_client_stub, ec2_client_stub):
...
+    # expect the bootstrapped config to have the custom security group...
+    # name and in bound rules
+    assert bootstrapped_config["provider"]["security_group"][
+        "GroupName"] == DEFAULT_SG_WITH_NAME_AND_RULES["GroupName"]
+    assert config["provider"]["security_group"][
+        "IpPermissions"] == CUSTOM_IN_BOUND_RULES
+
+    # Confirming boostrap config does not currently touch available node types.
+    assert bootstrapped_config["available_node_types"] == bootstrapped_config[
+        "available_node_types"]
```
Comment (ijrsvt) [inline python/ray/tests/aws/test_autoscaler_aws.py:260]: Should the second call be to `config`, not `bootstrapped_config`?

Comment (DmitriGekhtman) [reply python/ray/tests/aws/test_autoscaler_aws.py:260]: thanks, nice catch

**Review by ijrsvt — APPROVED** [review]
Looks good with that change!

**Pull request merged**